### ISSUE TITLE
Soft reset call not updating the change secret.

### ISSUE THREAD

**Issue by pmatthews05** [Azure-Samples/serverless-keyvault-secret-rotation-handling#8]
First of all, I want to thank you for such as smooth deployment to be able to test in my environment. It deployed as expected and without any hiccups.

I was able to get the function up and running without any issue, and could see in application insights [prefix]-serverless-ai-first streaming, with the second one sitting there.

![image](https://user-images.githubusercontent.com/6736509/162148846-9d97f06a-9c6f-453d-abd8-e910fc23e69d.png)

I updated the KeyVault with the second application insights key. This happened at 9:02:11AM 7th April 2022 (BST). The logic app clearly fired successfully at 9:04am (BST).
![image](https://user-images.githubusercontent.com/6736509/162151420-00a238b0-43ab-474f-b44e-55d4a636380a.png)

But the application insights has not swapped over. You can see the Logging event is at 8:07:40 AM, but this is GMT.
![image](https://user-images.githubusercontent.com/6736509/162151630-d2dc90db-8d3b-41ee-a481-d74b18f12d03.png)

Even calling the REST API from powershell (Using AZ Cli), it still doesn't pick up the change of secret.
```ps1
[CmdletBinding()]
param(
    [Parameter(Mandatory)]
    [string]$FunctionName,
    [Parameter(Mandatory)]
    [string]$resourceGroup
)
$InformationPreference = 'Continue'

Write-Information "Getting Function Name and Resource Group..."
Write-Information "FunctionName: $functionName"
Write-Information "ResourceGroupName: $resourceGroup"

Write-Information "Getting Azure Function Id"
$functionAppId = az functionapp show --name $functionName --resource-group $resourceGroup --query id --output tsv

$token = az account get-access-token --resource=https://management.azure.com --query accessToken --output tsv

$url = "https://management.azure.com$functionAppId/restart?api-version=2019-08-01&softRestart=true&synchronous=true"

$Headers = @{
    'Content-Type'  = 'application/json'
    'Authorization' = "Bearer $token"
}

Write-Information "Calling $url..."
Invoke-RestMethod -uri $url -Headers $Headers -Method POST 
```
As you can see my script runs without any errors, but it still stays as serverless-ai-first running the results.
![image](https://user-images.githubusercontent.com/6736509/162153369-84f2f231-ea87-4b0b-b3e5-c77df2b2f501.png)

The only way I can get it to swap, is updating something in the App Configurations. I created a new app configuration called update.
![image](https://user-images.githubusercontent.com/6736509/162154206-63a9ddc0-027c-4c7d-8492-149f1ea57b45.png)

As you can see from the below screenshot, after about 20-30 seconds, the switch happened.
![image](https://user-images.githubusercontent.com/6736509/162154697-ff57fe8f-c5de-45f3-aa9d-426c50a6fae7.png)

Clearly there isn't a problem with your solution. The problem is around the API call. But as @brandom-msft you work at Microsoft, you might be able to find out if this API has changed, or if there a different way to do this now.

**Comment by pmatthews05** [comment]
Some additional information I've just found. Looking at the Function App Activity log, the Restart Web App seems to be initiated twice by the logic app.  First status accepted, the second fails. Wondering if this is the reason why it's not updating.

![image](https://user-images.githubusercontent.com/6736509/162172947-e92a70ff-4b87-4921-a940-6a96c008b6fe.png)

With the following information:
```json
  "submissionTimestamp": "2022-04-07T09:48:01.1422431Z",
    "subscriptionId": "<Removed>",
    "tenantId": "<Removed>",
    "properties": {
        "statusMessage": "{\"status\":\"Failed\",\"error\":{\"code\":\"ResourceOperationFailure\",\"message\":\"The resource operation completed with terminal provisioning state 'Failed'.\",\"details\":[{\"code\":\"InternalServerError\",\"message\":\"There was an unexpected InternalServerError.  Please try again later.  x-ms-correlation-request-id: d8054efa-c175-4ce9-880c-5b20fdcce16e\"}]}}",
        "eventCategory": "Administrative",
        "entity": "/subscriptions/<Removed>/resourceGroups/pm5-serverless-sample/providers/Microsoft.Web/sites/pm5-serverless-functionapp",
        "message": "Microsoft.Web/sites/restart/action",
        "hierarchy": "<Removed>"
    },
    "relatedEvents": []
```

**Comment by pmatthews05** [comment]
Sorry for reference other people, you have all been a contributor to the article/code at some point. Just would like to get a conversation going about this issue.

@tyeth @polatengin @nzthiago & @devlead

**Comment by brandonh-msft** [comment]
Investigating for repro right now by just going thru deployment & setup. will report back if I find anything but have assigned to Matt for now

**Comment by brandonh-msft** [comment]
I believe we are now at the mercy of new behavior defined [here](https://docs.microsoft.com/en-us/azure/app-service/app-service-key-vault-references#rotation):
> When newer versions become available, such as with a rotation event, the app will automatically update and begin using the latest version within 24 hours. The delay is because App Service caches the values of the key vault references and refetches it every 24 hours

This sample could be modified to, instead of restarting the app, change the value of the appsetting to include the version from keyvault. When the secret is updated, then, as part of the Logic App's execution we should change the appsetting value (instead of restarting the app) which I found results in the desired behavior from my testing.

I'm going to close this issue since we've determined it's not a bug in the sample but rather new behavior from AppService. If the team wants to do the necessary work they can re-open and associate w/ commits/PRs or they can file a new feature request.
